File: PHP/program/Mahasiswa.php
<?php
// import file
require ('Sivitas.php');

// deklarasi kelas mahasiswa
class Mahasiswa extends Sivitas {
    // atribut private
    private $nim = '';
    private $prodi = '';
    private $fakultas = '';

    // konstruktor
    public function __construct( $nim = '', $prodi = '', $fakultas = '',$foto='' ) {
        $this->nim = $nim;
        $this->prodi = $prodi;
        $this->fakultas = $fakultas;
    }
 /* Getter dan Setter */
    public function set_nim($nim){
        $this->nim = $nim;
    }

    public function get_nim(){
        return $this->nim;
    }
    
    public function set_prodi($prodi){
        $this->prodi = $prodi;
    }

    public function get_prodi(){
        return $this->prodi;
    }
    
    public function set_fakultas($fakultas){
        $this->fakultas = $fakultas;
    }
    public function get_fakultas(){
        return $this->fakultas;
    }

}

?>

File: CPP/program/Mahasiswa.cpp
// import library
#include <iostream>
#include <string>
#include "Sivitas.cpp"

// menggunakan standard namespace
using namespace std;

// deklarasi kelas mahasiswa
class Mahasiswa : public Sivitas
{
    // atribut private
private:
    string nim, prodi, fakultas;

public:
    /* konstruktor */

    // konstruktor tanpa parameter
    Mahasiswa()
    {
        // set isi atribut secara default
        this->nim = "";
        this->prodi = "";
        this->fakultas = "";
    }
    // konstruktor dengan parameter
    Mahasiswa(string nim, string prodi, string fakultas)
    {
        // set isi atribut berdasarkan isi parameter
        this->nim = nim;
        this->prodi = prodi;
        this->fakultas = fakultas;
    }

    /* Getter dan Setter*/

    // Get nim
    string get_nim()
    {
        return this->nim;
    }

    // Set nim
    void set_nim(string nim)
    {
        this->nim = nim;
    }

    // Get prodi
    string get_prodi()
    {
        return this->prodi;
    }

    // Set prodi
    void set_prodi(string prodi)
    {
        this->prodi = prodi;
    }

    // Get fakultas
    string get_fakultas()
    {
        return this->fakultas;
    }

    // Set fakultas
    void set_fakultas(string fakultas)
    {
        this->fakultas = fakultas;
    }

    /* desktruktor */

    // default
    ~Mahasiswa() // desktruktor
    {
    }
};
